batched api / eti documentation

alphataubio opened the pull request:
- [ ] batched/dense
- [ ] batched/sparse
- [ ] eti

alphataubio commented:
@lucbv READY FOR REVIEW

**( Preview: https://alphataubio.com/kokkos-kernels/ )**

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+where:
+
+- :math:`I` is the identity matrix
+- :math:`u` is the Householder vector
```
I am not sure a Householder vector is really a thing, I would argue that it is more interesting to explain what it is with a formula such as `u_x=x-||x||e_0`

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+
+- :math:`I` is the identity matrix
+- :math:`u` is the Householder vector
+- :math:`\tau` is the Householder scalar
```
I am pretty sure a Householder scalar is not a thing, tau is defined as `tau = 2 / (u^hu)`

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+- :math:`I` is the identity matrix
+- :math:`u` is the Householder vector
+- :math:`\tau` is the Householder scalar
+- :math:`u^T` is the transpose of :math:`u`
```
No need for this

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+- :math:`u` is the Householder vector
+- :math:`\tau` is the Householder scalar
+- :math:`u^T` is the transpose of :math:`u`
+- :math:`u^H` is the conjugate transpose of :math:`u` (for complex matrices)
```
Also no need for this

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+Parameters
+==========
+
+:member: Team execution policy instance (only for team version)
```
No, this is a team policy member

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+==========
+
+:member: Team execution policy instance (only for team version)
+:u2: View containing the Householder vector
```
Not really: view containing the entries of `u` starting at the second element, so it's size is n-1 if u is of size n

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+        // Copy results back to host
+        Kokkos::deep_copy(A_host, A);
+        
+        // Verify results: Manually compute H*A and compare
```
This is not useful for an example, this is already implemented in our unit-tests with proper checks

lucbv commented on docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst:
```diff
@@ -0,0 +1,326 @@
+        // Copy results back to host
+        Kokkos::deep_copy(A_host, A);
+        
+        // Verify results for a few batches
```
Please remove everything below

lucbv reviewed (changes requested):
Here are some initial comment on the first few files, more to come.

alphataubio commented:
I've addressed all the reviewer comments from @lucbv:

## Fixed Issues:

### 1. batched-index.rst
- **Issue**: File was mostly wrong and misleading
- **Fix**: Removed the misleading content and returned to a simple placeholder structure

### 2. AddRadial.rst
- **Issue**: "Team execution policy instance" should be "Team policy member"
- **Fix**: ✅ Changed parameter description 
- **Issue**: create_mirror_view vs create_mirror
- **Fix**: ✅ Changed to create_mirror to keep separate allocation
- **Issue**: Extensive verification code not needed for examples
- **Fix**: ✅ Removed verification code, replaced with simple output to demonstrate function worked

### 3. ApplyHouseholder.rst
- **Issue**: "Householder vector" and "Householder scalar" terminology incorrect
- **Fix**: ✅ Fixed to proper mathematical definitions:
  - Householder vector: u = x - ||x||e_0
  - tau = 2 / (u^H u)
- **Issue**: Unnecessary explanations of u^T and u^H
- **Fix**: ✅ Removed redundant explanations
- **Issue**: "Team execution policy instance" should be "Team policy member"
- **Fix**: ✅ Changed parameter description
- **Issue**: u2 parameter description needed clarification
- **Fix**: ✅ Clarified as "entries starting at second element (size n-1 if u is of size n)"
- **Issue**: Extensive verification code not needed
- **Fix**: ✅ Removed verification code, replaced with simple output

All changes maintain the same functional examples while making them more concise and appropriate for documentation rather than unit tests, as requested.

alphataubio commented:
@lucbv please review again, let me know if anything else need polishing

lucbv commented:
We actually went ahead and started to manually write the batched algorithms documentation.
